packages/core/content-manager/admin/src/hooks/useDocument.ts

```diff
@@ -107,13 +117,31 @@ const useDocument: UseDocument = (args, opts) => {
     ...opts,
     skip: (!args.documentId && args.collectionType !== SINGLE_TYPES) || opts?.skip,
   });
+  const document = data?.data;
+  const meta = data?.meta;
 
   const {
     components,
     schema,
     schemas,
     isLoading: isLoadingSchema,
   } = useContentTypeSchema(args.model);
+  const isSingleType = schema?.kind === 'singleType';
+
+  const getTitle = (mainField: string) => {
+    if (mainField !== 'id' && document?.[mainField]) {
```

Comment:
I think it is better to move the comment before the conditions just to explain better the different scenarios, for example

`
// Use always the mainField if it's not an id
if (mainField !== 'id' && document?.[mainField]) {

`

Reply:
I don't really have a preference, if you prefer it this way I can update.